libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -589,7 +589,10 @@ void ContractLevelChecker::checkStorageSize(ContractDefinition const& _contract)
 	bigint size = 0;
 	for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts | ranges::views::reverse)
 		for (VariableDeclaration const* variable: contract->stateVariables())
-			if (!(variable->isConstant() || variable->immutable()))
+			if (
+				!(variable->isConstant() || variable->immutable()) &&
+				variable->referenceLocation() == VariableDeclaration::Location::Unspecified
```

Comment:
We should be checking this for transient variables as well. Currently it's hard to go over the limit, but we probably won't remember to update it once we allow reference types.

Reply:
Yeah, I was thinking about this too.
Added the check.